alot/db/utils.py

```diff
@@ -135,10 +135,20 @@ def _handle_signatures(original_bytes, original, message, params):
         if not mic_alg.startswith('pgp-'):
             raise MessageError(f'expected micalg=pgp-..., got: {mic_alg}')
 
-        # manual part extraction from original message necessary because
-        # parsing and re-serialising a Message isn't byte-perfect,
-        # which interferes with signature validation.
-        # See RFC 1847 section 2.1.
+        # RFC 3156 section 5 says that "signed message and transmitted message
+        # MUST be identical". We therefore need to validate against the message
+        # as it was originally sent.
+
+        # The transmitted content and therefore the signed content are using
+        # CRLF as line delimiter, but our eml file has most likely been
+        # converted to UNIX LF line ending in the local storage.
+        if b'\r\n' not in original_bytes:
```

Comment:
Offline discussion: This is horrible, but the RFC3156 does specify the data and headers (including MIME content headers) must be converted [to the canonical <CR><LF> sequence](https://tools.ietf.org/html/rfc3156#section-5). This is unfortunately correct.